### ISSUE TITLE
Muon user data directories not available for import on Ubuntu

### ISSUE THREAD

**Issue by garrettr** [brave/brave-browser#3130]
## Description
After upgrading from Muon to brave-browser on Ubuntu, user reports they cannot import their Muon profile data because there are no Muon user data directories listed in the menu of import data source options.

Original report via [Twitter](https://twitter.com/majagray75/status/1088640072264474624).

## Steps to Reproduce
1. Use Muon on Ubuntu
2. Install brave-browser via Snap (based on [tweet](https://twitter.com/majagray75/status/1089728463949295616))
3. Open brave-browser.
4. Attempt to **Import Bookmarks and Settings...**
5. Click to reveal the drop-down menu in the **Import bookmarks and settings** dialog

## Actual result:
No Muon user data directories are listed as being available for import.

## Expected result:
All Muon user data directories that exist on the system and can be imported should be listed.

## Reproduces how often: 
Probably an intermittent issue; however, was reported externally so I am not sure how difficult it is to reproduce.

## Brave version (brave://version info)
Latest Brave version available via Snap on Ubuntu.

### Additional Information
This is almost certainly an issue with the following code: https://github.com/brave/brave-core/blob/461df5326f471b6048a984de420f03afe01f3038/common/importer/brave_importer_utils_linux.cc#L12-L25.

1. The current code assumes that if a `brave` binary is available via Snap, then the active user data directory is available in the `.config/` symlinked under `$HOME/snap/brave/current`. This assumption may be faulty.
2. There may be multiple Muon user data directories available from different Snap-installed versions of Brave, e.g. under `~/snap/brave/<version number>/.config/brave`. Each of these should be available for import, as long as they are importable—it's possible the user data directories from some earlier versions might not be.
3. The current code ignores the user data directory used by Muon when it is installed via apt (`~/.config/brave`) if it finds one from a Snap-installed version. This choice was intended to avoid confusing users, but it now appears to be too clever for its own good. It's probably better just to list all available Muon user data directories and let the user sort out which one they wish to import from.

**Comment by payingattention** [comment]
Same issue, the official program build acts like nothing is happening, and is my bank/money equivalent. Definitiely `git blame` that "Snapcraft". Reproduced the problem.

What are we aiming for as a cryptofinancial track record (if forcing [even suggesting the `snap` idea is *dangerous* in our example] users to use "snaps" often (often meaning insofar nearly standardizing that the problem is the "snap autoupgrade feature" itself pointblank), by nature of documented Snapcraft ideology, repeats the same problem and we do not immediately socially/programmatically address the issue)?

Add to that, there is extremely and outrageously limited documentation, and it is often the sparse *official* docs that are "just linking" to very informally and disorganizedly answered questions on the official community help section, on the official governing domain directing people what to do about the heavily pushed ("automatically pushed"... id argue that is a human being mass code shooting an idea from the Brave domain that we are caling "automatic") major complete system name/program change, even finding a timestamped reference to the names/programs changing formally is difficult to find in *written* form by any Brave authority, like it never happened.

I can barely even word the problem. "Git blame" the Snap or Brave and is that important or should, the second we saw what the program did, I/we just say/report?

I found the keywords, but how many users are just screwed and just panic (and now "just broke", like that's really about being just broke...), are we using/justifying analytics (I voluntarily set to give my analytics to Brave for free and can prove the config, so if we track how many brave|snap bank balances are lingering too) for that too?
